src/Compilers/Core/Portable/Syntax/AbstractWarningStateMap.cs

```diff
@@ -87,5 +87,13 @@ public int CompareTo(WarningStateMapEntry other)
                 return this.Position - other.Position;
             }
         }
+
+        protected class WarningStateMapEntryComparer : System.Collections.IComparer
```

Comment:
1. No reason it's public. I think I just matched the visibility of MapEntry. It can definitely be private.
2. Roger
3. Can you point me to the codebase's preferred singleton pattern? Or anything goes?